openwisp_firmware_upgrader/tests/test_admin.py

```
@@ -510,394 +516,431 @@ def test_admin_multitenancy(self):
-
-
-@mock.patch(_mock_upgrade, return_value=True)
-@mock.patch(_mock_connect, return_value=True)
 class TestAdminTransaction(
     BaseTestAdmin, AdminActionPermTestMixin, TransactionTestCase
 ):
+    _mock_upgrade = "openwisp_firmware_upgrader.upgraders.openwrt.OpenWrt.upgrade"
+    _mock_connect = "openwisp_controller.connection.models.DeviceConnection.connect"
+
+    @mock.patch(_mock_upgrade, return_value=True)
     def test_upgrade_selected_action_perms(self, *args):
```

Ruff (0.14.10)

Unused method argument: `args`

(ARG002)